src/Compiler/CompilerPasses.cpp

```diff
@@ -153,6 +153,13 @@ void addONNXToMLIRPasses(mlir::PassManager &pm, bool targetCPU,
   // function and just before instrumentation.
   pm.addPass(createSetONNXNodeNamePass());
 
+  // Add a CSE pass to delete unreachable ops.
+  // This is useful when a model, especially manually modified one, is dumped
+  // with --EmitONNXIR.
+  // This pass is inserted before the instrumentation pass, which may add
+  // instrumentation ops with external sideeffect.
+  pm.addPass(mlir::createCSEPass());
```

Comment:
I wonder if `symbol-dce` is more appropriate?: https://mlir.llvm.org/docs/Passes/#-symbol-dce